app/uk/gov/hmrc/exports/services/SubmissionService.scala

```diff
@@ -52,54 +53,65 @@ class SubmissionService @Inject() (
       case _                                                            => Future.successful(NotFound)
     }
 
-  // Fetch first page of submissions for the first StatusGroup containing submissions.
-  def fetchFirstPage(eori: String, limit: Int): Future[PageOfSubmissions] = {
-    def loopOnGroups(f: Set[String] => Future[Seq[Submission]])(cond: Seq[Submission] => Boolean)(seq: List[Set[String]]): Future[Seq[Submission]] =
+  def fetchFirstPage(eori: String, statusGroups: Seq[StatusGroup], limit: Int): Future[PageOfSubmissions] = {
+    // When multiple StatusGroup(s) are provided, the fetch proceeds in sequence group by group.
+    // When a page/batch of Submissions (the first page actually, 1 to limit) is found for a group,
+    // that page is the result of the fetch. The successive groups are ignored.
+
+    @nowarn("msg=match may not be exhaustive")
+    def loopOnGroups(f: StatusGroup => Future[Seq[Submission]])(cond: Seq[Submission] => Boolean)(seq: Seq[StatusGroup]): Future[Seq[Submission]] =
       seq match {
         case head :: tail => f(head).filter(cond).recoverWith { case _: Throwable => loopOnGroups(f)(cond)(tail) }
         case Nil          => Future.successful(Seq.empty)
       }
 
     for {
       submissions <- loopOnGroups(submissionRepository.fetchFirstPage(eori, _, limit))(_.nonEmpty)(statusGroups)
-      setOfEnhancedStatuses = submissions.headOption.flatMap(_.latestEnhancedStatus.map(fromEnhancedStatus)).getOrElse(Set.empty)
-      totalSubmissionsInGroup <- submissionRepository.countSubmissionsInGroup(eori, setOfEnhancedStatuses)
+      totalSubmissionsInGroup <- countSubmissionsInGroup(eori, submissions)
     } yield {
       val statusGroup = submissions.headOption.flatMap(_.latestEnhancedStatus.map(toStatusGroup)).getOrElse(SubmittedStatuses)
       PageOfSubmissions(statusGroup, totalSubmissionsInGroup, submissions)
     }
   }
 
-  // Fetch first page of submissions for a specific StatusGroup
   def fetchFirstPage(eori: String, statusGroup: StatusGroup, limit: Int): Future[PageOfSubmissions] =
-    fetchPage(eori, statusGroup, submissionRepository.fetchFirstPage(eori, _, limit))
+    // Fetch first page of submissions for the provided StatusGroup
+    fetchPage(eori, statusGroup, () => submissionRepository.fetchFirstPage(eori, statusGroup, limit))
 
   def fetchPage(eori: String, statusGroup: StatusGroup, fetchPageData: FetchSubmissionPageData): Future[PageOfSubmissions] = {
-    val fetchSubmissions = (setOfEnhancedStatuses: Set[String]) =>
+    val fetchFunction = () =>
       fetchPageData.datetimeForPreviousPage.fold {
         fetchPageData.datetimeForNextPage.fold {
           fetchPageData.page.fold {
-            submissionRepository.fetchLastPage(eori, setOfEnhancedStatuses, fetchPageData.limit)
+            // datetimeForPreviousPage, datetimeForNextPage and page NOT provided => fetching the last page
+            submissionRepository.fetchLastPage(eori, statusGroup, fetchPageData.limit)
```

Comment:
No, no. We want to return the last page when the `page` parameter is not provided, since we know which page is the first one (`page=1`) but don't know which is the last one.